src/game/hud.cpp

```diff
@@ -1355,13 +1383,33 @@ namespace hud
                 }
                 if(crosshairhitspeed && totalmillis-game::focus->lasthit <= crosshairhitspeed)
                 {
-                    vec c2(1, 1, 1);
-                    if(hitcrosshairtone) skewcolour(c2.r, c2.g, c2.b, hitcrosshairtone);
-                    else c2 = c;
-                    drawpointertex(getpointer(POINTER_HIT, game::focus->weapselect), cx-cs/2, cy-cs/2, cs, c2.r, c2.g, c2.b, crosshairblend*hudblend);
+                    hit = true; 
+                    if (hitcrosshairtone)
+                    {
+                        skewcolour(color_hit.r, color_hit.g, color_hit.b, hitcrosshairtone);
+                    }
+                    else
+                    {
+                        if (!(crosshairweapons&2)) // if the player has enabled custom crosshair colors
```

Comment:
More in-line comments. Also, this is an implicit check. The binary AND returns a numeric value, so please compare that to a number explicitly. This is just not very readable.

Reply:
that's the only way it worked, I'm don't know how else I could've done it as I never worked with these binary operators before